src/BeliefTypes.jl

```diff
@@ -41,3 +41,41 @@ mutable struct NBPMessage <: Singleton
 end
 
 ### SOME CONVERGENCE REQUIRED ^^^
+
+#DEV NOTE it looks like it can be consolidated into one type
+# if we can pass messages similar to EasyMessage:
+# pts::Array{Float64,2}
+# bws::Array{Float64,1}
+# option a tuple
+# bellief::Dict{Symbol, NamedTuple{(:vec, :bw, :inferdim),Tuple{Array{Int64,1},Array{Int64,1},Float64}}}
+# or an extra type
+# or the MsgPrior/PackedMessagePrior, depending on the serialization requirement of the channel
+# but I would think only one message type
+
+# mutable struct NBPMessage <: Singleton
+#   status::Symbol # Ek kort die in die boodskap
+#   p::Dict{Symbol, EasyMessage}
+# end
+
+struct TreeBelief
+  val::Array{Float64,2}
+  bw::Array{Float64,2}
+  inferdim::Float64
+  # manifolds::T TODO ? JT not needed as you have the variable with all the info in it?
```

Comment:
yeah, agree that we have the manifolds from softtypes in the variables themselves, which should be plenty to get the ball rolling.